pkg/controller/telemetry/metric.go

```diff
@@ -577,14 +577,22 @@ func (m *MetricController) updatePrometheusMetric() {
 	// delete metrics
 	workloadDeleteLength := len(deleteWorkload)
 	serviceDeleteLength := len(deleteService)
+	// Creating a copy reduces the amount of time spent adding locks in the programme
+	workloadLock.Lock()
+	workloadCopy := deleteWorkload[:workloadDeleteLength]
```

Comment:
It's better to change the name, because in fact no copy occures, just a slice split.